source/AI.cpp

```diff
@@ -2654,6 +2663,41 @@ void AI::DoScatter(Ship &ship, Command &command)
 
 
 
+bool AI::DoSecretive(Ship &ship, Command &command)
+{
+	shared_ptr<Ship> scanningShip;
+	// Figure out if any ship is currently scanning us. If that is the case, move away from it.
+	for(auto otherShip : GetShipsList(ship, false))
+		if(!ship.GetGovernment()->Trusts(otherShip->GetGovernment()) &&
+				otherShip->Commands().Has(Command::SCAN) &&
+				otherShip->GetTargetShip() == ship.shared_from_this() &&
+				!otherShip->IsDisabled() && !otherShip->IsDestroyed())
+			scanningShip = make_shared<Ship>(*otherShip);
+
+	if(scanningShip)
+	{
+		Point scanningPos = scanningShip->Position();
+		Point pos = ship.Position();
+
+		double cargoDistance = 100. * sqrt(scanningShip->Attributes().Get("cargo scan power"));
+		double outfitDistance = 100. * sqrt(scanningShip->Attributes().Get("outfit scan power"));
+
+		double maxScanRange = cargoDistance > outfitDistance ? cargoDistance : outfitDistance;
+		double distance =scanningPos.Distance(pos);
+
+		// If he can scan us we need to evade.
+		if(distance < maxScanRange)
+		{
+			Point away = scanningPos + Angle(pos - scanningPos).Unit() * (maxScanRange + 50.);
```

Comment:
You would then need to square root the thing here, ~or just don't bother and make sure the ship really is out of range.~

Reply:
*Scans a ship*
ship: I shall run and never look back!
yeah best to do one sqrt instead of 3